File: 1773/1773.dart
int countMatches(List<List<String>> items, String ruleKey, String ruleValue) {
  int flag = 0;

  int typeIndex = ["type", "color", "name"].indexOf(ruleKey);

  items.forEach((element) {
    if (element[typeIndex] == ruleValue) {
      flag++;
    }
  });

  return flag;
}

main() {
  final items = [
    ["phone", "blue", "pixel"],
    ["computer", "silver", "lenovo"],
    ["phone", "gold", "iphone"]
  ];

  final ruleKey = "color";
  final ruleValue = "silver";

  print(countMatches(items, ruleKey, ruleValue));
}

File: 1773/1773.js
var countMatches = function(items, ruleKey, ruleValue) {
    var flag = 0;

    var typeIndex = ["type", "color", "name"].indexOf(ruleKey); 

    items.array.forEach(element => {
        if(element[typeIndex] == ruleValue) {
            flag++;
        }
    });

    return flag;
}